pkg/controller/test/context_builder.go

```diff
@@ -157,6 +158,15 @@ func (b *Builder) Init() {
 	apiutil.Clock = b.Context.Clock
 }
 
+// InitWithRESTConfig() will call builder.Init(), then assign an initialised
+// RESTConfig with a `cert-manager/unit-test` User Agent.
+func (b *Builder) InitWithRESTConfig() {
```

Comment:
I quite like the "options" pattern for this. I think I've seen it elsewhere in the codebase.

Would it make sense for us to implement that here? (Not necessarily now, but if we were to add more alternatives to `Init`)

Reply:
Yeah, if we needed to expose more options to the `Init` function, then some kind of options pattern makes sense. The function here came out of the comment [here](https://github.com/jetstack/cert-manager/pull/4638#discussion_r761804759).